src/internal_events/amqp.rs

```diff
@@ -0,0 +1,74 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct AmqpEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for AmqpEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "Received one event.", internal_log_rate_secs = 10);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_events_total", 1);
+        counter!("processed_bytes_total", self.byte_size as u64);
+    }
+}
+
+#[derive(Debug)]
+pub struct AmqpConsumerFailed {
+    pub error: lapin::Error,
+}
+
+impl InternalEvent for AmqpConsumerFailed {
+    fn emit_logs(&self) {
+        error!(message = "Failed to consume.", error = ?self.error);
```

Comment:
```suggestion
        error!(message = "Failed to consume.", error = ?self.error,  internal_log_rate_secs = 10);
```

We generally include rate limits to avoid flooding user logs.